src/bun.js/node/node_fs_watcher.zig

```diff
@@ -403,6 +404,32 @@ pub const FSWatcher = struct {
 
                             current_task.append(relative_path, event_type, .destroy);
 
+                            const full_path = bun.default_allocator.dupeZ(u8, path_slice) catch unreachable;
+                            const is_watched = blk: {
+                                for (this.file_paths.slice()) |path| {
+                                    if (std.mem.eql(u8, path, full_path)) {
+                                        break :blk true;
+                                    }
+                                }
+                                break :blk false;
+                            };
+
+                            // Add new files or directories to the watch list.
+                            if (!is_watched) {
+                                this.file_paths.push(bun.default_allocator, full_path) catch unreachable;
+                                const fs_info = fdFromAbsolutePathZ(full_path) catch null;
+                                if (fs_info) |info| {
+                                    if (info.is_file) {
+                                        this.default_watcher.?.addFileAssumeLocked(info.fd, full_path, file_hash, options.Loader.file, 0, null, false) catch unreachable;
+                                        current_task.append(relative_path, .change, .destroy);
```

Comment:
Add this because when a new file is created, two events (`rename`, `change`) should be triggered.

Reply:
`relative_path` is created by `bun.default_allocator.dupe`.